lib/core/client.js

```diff
@@ -516,6 +516,9 @@ class Parser extends HTTPParser {
 
     const request = client[kQueue][client[kRunningIdx]]
 
+    // TODO (fix): Why does this assertion fail?
+    // assert(!client[kHeadersTimeout] || this.timeout)
```

Comment:
`this.timeout` must be getting set to `null` before getting here. Yeah, it's puzzling